src/gpu/nvidia/cudnn_inner_product.cpp

```diff
@@ -34,34 +35,28 @@ status_t cudnn_inner_product_fwd_t::execute(const exec_ctx_t &ctx) const {
             = utils::downcast<nvidia::sycl_cuda_stream_t *>(ctx.stream());
 
     return cuda_stream->interop_task([&](::sycl::handler &cgh) {
-        using scratch_acc_t = ::sycl::accessor<uint8_t, 1,
-                ::sycl::access::mode::read_write, sycl::compat::target_device>;
-        using read_acc_t = ::sycl::accessor<uint8_t, 1,
-                ::sycl::access::mode::read, sycl::compat::target_device>;
-        auto src_acc = CTX_IN_ACCESSOR(DNNL_ARG_SRC);
-        auto wei_acc = CTX_IN_ACCESSOR(DNNL_ARG_WEIGHTS);
-        std::shared_ptr<read_acc_t> bias_acc;
-        if (pd()->with_bias()) {
-            bias_acc = std::make_shared<read_acc_t>(
-                    CTX_IN_ACCESSOR(DNNL_ARG_BIAS));
-        }
-        auto dst_acc = CTX_OUT_ACCESSOR(DNNL_ARG_DST);
-        std::shared_ptr<scratch_acc_t> ip_scratch_acc;
-        std::shared_ptr<scratch_acc_t> spacial_scratch_acc;
-        std::shared_ptr<scratch_acc_t> scaled_bias_scratch_acc;
+        cgh.depends_on(cuda_stream->get_deps());
+
+        auto arg_src = CTX_IN_SYCL_MEMORY(DNNL_ARG_SRC);
+        auto arg_wei = CTX_IN_SYCL_MEMORY(DNNL_ARG_WEIGHTS);
+        auto arg_bias = CTX_IN_SYCL_MEMORY(DNNL_ARG_BIAS);
+        auto arg_dst = CTX_OUT_SYCL_MEMORY(DNNL_ARG_DST);
+
+        sycl_memory_arg_t<::sycl::access::mode::read_write> arg_ip_scratch;
+        sycl_memory_arg_t<::sycl::access::mode::read_write> arg_spacial_scratch;
+        sycl_memory_arg_t<::sycl::access::mode::read_write>
+                arg_scaled_bias_scratch;
         if (pd()->inner_product_impl_->ip_using_scratchpad()) {
-            ip_scratch_acc = std::make_shared<
-                    scratch_acc_t>(CTX_SCRATCH_ACCESSOR(
-                    memory_tracking::names::key_iprod_int_dat_in_acc_dt));
+            arg_ip_scratch = CTX_SCRATCH_SYCL_MEMORY(
+                    memory_tracking::names::key_iprod_int_dat_in_acc_dt);
         }
         if (pd()->inner_product_impl_->need_to_transform_filter()) {
-            spacial_scratch_acc = std::make_shared<scratch_acc_t>(
-                    CTX_SCRATCH_ACCESSOR(memory_tracking::names::key_none));
+            arg_spacial_scratch
+                    = CTX_SCRATCH_SYCL_MEMORY(memory_tracking::names::key_none);
```

Comment:
With [this ](https://github.com/oneapi-src/oneDNN/pull/1263/files#r810377333) minor tweak we can do this kinds of things unconditionally.